### ISSUE TITLE
Flow variance syntax

### ISSUE THREAD

**Issue by zertosh** [gajus/eslint-plugin-flowtype#137]
Flow 0.34.0 (upcoming release) is [introducing](https://flowtype.org/blog/2016/10/04/Property-Variance.html) syntax to annotate variance. The support was just added to [babylon@6.13.0](https://github.com/babel/babylon/blob/v6.13.0/CHANGELOG.md#v6130-2016-10-21). The new syntax clashes with `flowtype/space-after-type-colon` and maybe others.

Here's one I ran into:

```
type Foo = {
  +bar: () => void,
};
```

With:

```
    'flowtype/space-after-type-colon': [1, 'always'],
    'flowtype/space-before-type-colon': 1,
```

The `+` triggers `flowtype/space-after-type-colon`. Adding a space, still triggers the rule but then also triggers `flowtype/space-before-type-colon`. It's perfectly reasonable to want to have no space after the `+` (a lot like generic brackets).

cc: @nmote

**Comment by danharper** [comment]
👍 I've got a branch for `space-after-type-colon`, will add `space-before-type-colon` in the morning.

**Comment by danharper** [comment]
Released as 2.21.0 :)

**Comment by zertosh** [comment]
Just tried it! Thanks @danharper!
